apps/aecore/src/aec_sync.erl

```diff
@@ -84,29 +84,62 @@ compare_ping_objects(RemoteUri, Local, Remote) ->
         {G, G} ->
```

Comment:
I understand that the base64/base58 encoding is not part of the business logic, it is rather a type that should (ideally) be handled inside swagger.

May I suggest that the base64/base58 encoding/decoding is moved in `aehttp_dispatch_ext:handle_request('Ping', ...)` (so away from aec_sync), and that HTTP status code 400 (with descriptive error reason) is returned. In this was two genesis objects where (for any reason) one is base64 and the other base58 (or any other encoding/decoding quirk) is detected as bad client request format - rather than `different_genesis_block`.

Reply:
That is not exactly the right place to handle it. There are server and client side. What we should do is to handle with pure data in aec_syn and peer and only encode and decode in http request/response. The request and response for the client side are in aeu_http_client. Incomping pings do get to aehttp_dispatch_ext... but can be filtered before.